### PR TITLE
Adding multichain support for solidity code

### PR DESCRIPTION
**Pull request by ferencdg**
Major changes

* making the strategy and portfolio base classes work with any ERC-20 deposit tokens: investment tokens now have a fixed 6 decimal precision on all chains
* added native swapping methods
* added Venus Oracle

Minor changes

* making constants better typed
* adding BSC related constants
* added venus dependencies

### CODE REVIEW

File: contracts/dependencies/swap/IUniswapV2LikeRouter.sol [hunk 1]
```diff
@@ -5,25 +5,55 @@ interface IUniswapV2LikeRouter {
...
-        address[] memory path,
+        address[] calldata path,
         address to,
         uint256 deadline
     ) external returns (uint256[] memory amounts);
 
     function swapTokensForExactTokens(
         uint256 amountOut,
         uint256 amountInMax,
-        address[] memory path,
```
Comment (ferencdg) [inline contracts/dependencies/swap/IUniswapV2LikeRouter.sol:16]: We are using those interfaces to call external contracts, and even whether it is declared memory or calldata, solidity compiler will encode the 'path' to calldata format. Pretty much all the projects are checked are using 'calldata' here. (even us in our own Brokkr interfaces)

@jihoonsong let's talk about this offline if you don't agree

File: .openzeppelin/avalanche.json [hunk 2]
```diff
@@ -1101,7 +1101,7 @@
           },
           "t_enum(SwapServiceProvider)7954": {
             "label": "enum StrategyBaseUpgradeable.SwapServiceProvider",
-            "members": ["TraderJoe"],
```
Comment (ferencdg) [inline .openzeppelin/avalanche.json:1104]: Tricking the upgrade plugin into skipping the storage check for the enums.

**Comment by ferencdg** [comment]
Merging it now, as I go on a long vacation and people might need this code in the main. Leave your comments here anyway and will check it, after coming back.

File: contracts/dependencies/traderjoe/ITraderJoeLBRouter.sol [hunk 3]
```diff
@@ -29,8 +29,8 @@ interface ITraderJoeLBRouter {
         uint16 binStep,
         uint256 amountXMin,
         uint256 amountYMin,
-        uint256[] memory ids,
-        uint256[] memory amounts,
+        uint256[] calldata ids,
```
Comment (jihoonsong) [inline contracts/dependencies/traderjoe/ITraderJoeLBRouter.sol:32]: FYI, TraderJoe V2 uses memory.

File: contracts/strategies/dnsVectorStrategy/DnsVectorStrategy.sol [hunk 4]
```diff
@@ -29,12 +29,12 @@ contract DnsVectorStrategy is
 
     // solhint-disable-next-line const-name-snakecase
     string public constant trackingName =
-        "brokkr.dns_vector_strategy.dns_vector_strategy_v1.1.0";
+        "brokkr.dns_vector_strategy.dns_vector_strategy_v1.1.1";
```
Comment (jihoonsong) [inline contracts/strategies/dnsVectorStrategy/DnsVectorStrategy.sol:32]: Then both Stargate and TraderJoe have to update their version as well, I think

File: contracts/common/libraries/SwapServiceLib.sol [hunk 5]
```diff
@@ -11,9 +12,10 @@ import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeab
 error InvalidSwapServiceProvider();
 
 enum SwapServiceProvider {
-    TraderJoe,
-    TraderJoeV2,
-    Pangolin
+    AvalancheTraderJoe,
```
Comment (jihoonsong) [inline contracts/common/libraries/SwapServiceLib.sol:15]: I think constants/SwapServices.json also needs to be modified accordingly.

**Pull request merged**